drivers/adc/adc_stm32.c

```diff
@@ -216,6 +216,35 @@ static bool init_irq = true;
 #endif
 
 #ifdef CONFIG_ADC_STM32_DMA
+static void adc_stm32_enable_dma_support(ADC_TypeDef *adc)
+{
+	/* Allow ADC to create DMA request and set to one-shot mode as implemented in HAL drivers */
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+
+#if defined(ADC_VER_V5_V90)
+	if (adc == ADC3) {
+		LL_ADC_REG_SetDMATransferMode(adc,
+			ADC3_CFGR_DMACONTREQ(LL_ADC_REG_DMA_TRANSFER_LIMITED));
+		LL_ADC_EnableDMAReq(adc);
+	} else {
+		LL_ADC_REG_SetDataTransferMode(adc,
+			ADC_CFGR_DMACONTREQ(LL_ADC_REG_DMA_TRANSFER_LIMITED));
+	}
+#elif defined(ADC_VER_V5_X)
+	LL_ADC_REG_SetDataTransferMode(adc, LL_ADC_REG_DMA_TRANSFER_LIMITED);
+#else
+#error "Unsupported ADC version"
+#endif
+
+#else /* defined(CONFIG_SOC_SERIES_STM32H7X) */
+
+	/* Default mechanism for other MCUs */
+	LL_ADC_REG_SetDMATransfer(adc, LL_ADC_REG_DMA_TRANSFER_LIMITED);
```

Comment:
A quick look through the LL shows that the macro `LL_ADC_REG_DMA_TRANSFER_LIMITED` does not exist for F1 and F37x.
For those, the choice is between `LL_ADC_REG_DMA_TRANSFER_NONE` or `LL_ADC_REG_DMA_TRANSFER_UNLIMITED`.
A `#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_adc)` should be added for these cases. What we do inside is up to you: unlimited seems to imply a circular buffer, which may be hard to add, so maybe we can just not support this case for now and return an error.

Reply:
I would prefer to error as "not supported", as I do not have hardware or time to investigate all potential edge cases at the moment. And my idea was that my code as-is doesn't compile then it already implies "not supported", but I can make this explicit.

Reply:
Added an explicit error